server/server.js

```diff
@@ -22,12 +22,206 @@ app.use(express.static(reactStaticDir));
 app.use(express.static(uploadsStaticDir));
 app.use(express.json());
 
-app.get('/api/hello', (req, res) => {
-  res.json({ message: 'Hello World!' });
+app.get('/api/public/Tables/Recipes', async (req, res, next) => {
+  try {
+    const sql = `
+      select *
+        from "Recipes";
+    `;
+    const result = await db.query(sql);
+    const recipes = result.rows;
+    res.json(recipes);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.get('/api/public/Tables/uri', async (req, res, next) => {
+  try {
+    const { uri } = req.body;
+    const sql = `
+      select *
+        from "Recipes"
+        where "uri" = $1;
+    `;
+    const params = [uri];
+    const result = await db.query(sql, params);
+    const recipeWithId = result.rows[0];
+    res.json(recipeWithId);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.get('/api/public/Tables/Users', async (req, res, next) => {
+  try {
+    const sql = `
+      select *
+        from "Users";
+    `;
+    const result = await db.query(sql);
+    const users = result.rows;
+    res.json(users);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.get('/api/public/Tables/Favorites', async (req, res, next) => {
+  try {
+    const sql = `
+      select *
+        from "Favorites"
+        join "Recipes" using ("recipeId");
+    `;
+    const result = await db.query(sql);
+    const favorites = result.rows;
+    favorites.forEach((recipe) => {
+      fixIngredients(recipe);
+    });
+    res.json(favorites);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.post('/api/public/Tables/Favorites/:uri', async (req, res, next) => {
+  try {
+    const { uri } = req.params;
+    const sql = `
+      select "recipeId"
+        from "Recipes"
+        where "uri" = $1;
+    `;
+    const params = [atob(uri)];
```

Comment:
I'm curious why you convert the `uri` to binary? Isn't it a string already?

Reply:
The `uri` link was causing issues and not letting the fetch `post` go through properly.